src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AnnotationFieldDefHandler.java

```diff
@@ -0,0 +1,100 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.indentation;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Handler for annotation field definitions.
+ * Annotation field defintion closely resemble the method defintions and therefore extends
+ * MethodDefHandler.
+ *
+ * @author Vikramaditya Kukreja
+ */
+public class AnnotationFieldDefHandler extends MethodDefHandler {
+    /**
+     * Construct an instance of this handler with the given indentation check,
+     * abstract syntax tree, and parent handler.
+     *
+     * @param indentCheck   the indentation check
+     * @param ast           the abstract syntax tree
+     * @param parent        the parent handler
+     */
+    public AnnotationFieldDefHandler(IndentationCheck indentCheck,
+                                     DetailAST ast, AbstractExpressionHandler parent) {
+        super(indentCheck, ast, parent);
+    }
+
+    @Override
```

Comment:
@kukreja-vikramaditya I removed all methods from this class except the constructor.
The only differences in test I saw was message changing from `annotation field def type` to `annotation field def modifier`. There was no differences in requested indentations or missing violations.

Please explain why we need all this code if existing code handles everything fine outside of small name difference.